### PR TITLE
Bugfix/gh 150 externalize app identifier

### PR DESCRIPTION
**Pull request by stepio**
Implementation of a fix for issue https://github.com/drallgood/jpasskit/issues/150. Also a bunch of further technical improvements as a bonus.

Let me know if I missed anything.

### CODE REVIEW

File: jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java [hunk 1]
```diff
@@ -33,45 +34,55 @@
 import com.turo.pushy.apns.util.concurrent.PushNotificationFuture;
 import com.turo.pushy.apns.util.concurrent.PushNotificationResponseListener;
 
+import org.apache.commons.lang3.tuple.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import static de.brendamour.jpasskit.util.Assert.state;
```
Comment (drallgood) [inline jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: Not a big fan of static imports. Makes testing much more cumbersome.

File: jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java [hunk 2]
```diff
@@ -33,45 +34,55 @@
...
     private static final Logger LOGGER = LoggerFactory.getLogger(PKSendPushNotificationUtil.class);
     private static final String EMPTY_PUSH_JSON_STRING = "{}";
     private static final int POOL_SIZE_DEFAULT = 10;
 
     private ApnsClient client;
+    private Set<String> topics;
 
-    public PKSendPushNotificationUtil(final String pathToP12, final String passwordForP12) throws IOException {
-        this(pathToP12, passwordForP12, POOL_SIZE_DEFAULT);
+    public PKSendPushNotificationUtil(String keyStorePath, char [] keyStorePassword) throws IOException {
```
Comment (drallgood) [inline jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: Any reason why we can't keep using the pkcs12 file? That's what you export from the macOS Keychain so it would be easiest for everyone to use.
Either way we would need to deprecate the API and offer both options for a while

Comment (stepio) [reply jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: I just renamed the variables to keep them self-descriptive. I call it self-descriptive because we use them to init `KeyStore` entity in Java (according to the existing functionality, not me invention).

Also I see p12 as just a specific format of a more general "keystore" concept, so that's why I prefer this name.

So to sum up:
1. No functional changes or API changes, just renaming - we still work with p12 files. I even have tests to prove this.
2. I may be wrong, so if you insist that it's not a good idea, I'll revert this renaming back.

File: jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java (hunk 1 above)
Comment (stepio) [reply jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: When I see "testing", I think about unit tests. Is it what you meant? Cause I don't know of such limitations - I'd appreciate if you could explain them to me.

If you tell about debugging and overall code readability - this is quite a subjective opinion, but I'm fine with following your code style. As I understand, this code style accepts static imports in test code, but does not recommend it for the main one.

Please correct me if I'm wrong.

**Comment by stepio** [comment]
I think that's a misunderstanding. I was referring to `keyStore` not in terms of source files, but talking about the resulting entities.
Here is a short snippet of the code before my changes (fro the previous release):
https://github.com/drallgood/jpasskit/blob/jpasskit-0.1.0/jpasskit/src/main/java/de/brendamour/jpasskit/signing/PKSigningInformationUtil.java
```java
    public KeyStore loadPKCS12File(final InputStream inputStreamOfP12, final String password) throws KeyStoreException, NoSuchAlgorithmException,
            CertificateException, IOException {
        if (inputStreamOfP12 == null) {
            throw new IllegalArgumentException("InputStream of key store must not be null");
        }
        KeyStore keystore = KeyStore.getInstance("PKCS12");

        keystore.load(inputStreamOfP12, password.toCharArray());
        return keystore;
    }
```

As you see, we use the method's argument to construct `KeyStore` from p12 file. And basically it is a keystore as it holds both certificate and private key for it.

In short the idea of this PR:
1. Extract basic methods for dealing with certificates into a common utility class.
2. "Borrow" a method for extracting topic names from `pushy` project and put it into the same utility class.
3. Reuse new utilities whenever is possible.
4. Fix the original issue with above given "building blocks".
5. Cover the changes with unit tests.

File: jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java (hunk 1 above)
Comment (drallgood) [reply jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: static imports can't be easily mocked in unit tests making it hard to test things in isolation (=> unit tests)

File: jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java (hunk 2 above)
Comment (drallgood) [reply jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: ok. that makes sense

File: jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java (hunk 1 above)
Comment (stepio) [reply jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: Mmm... I understand your idea - I did not have such issues myself, but it makes sense.

So... we allow static imports only for test code, right? No need to update test files, I mean.

Comment (drallgood) [reply jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: yeah

Comment (stepio) [reply jpasskit/src/main/java/de/brendamour/jpasskit/apns/PKSendPushNotificationUtil.java]: Done!

**Comment by stepio** [comment]
Requested assistance from @codacy-bot support to find out why their validation is stuck.
